pkg/platform/api/buildplanner/model/buildplan.go

```diff
@@ -302,22 +302,43 @@ func IsErrorResponse(errorType string) bool {
 		errorType == ComitHasNoParentErrorType
 }
 
+type CommitError struct {
+	Type                   string
+	Message                string
+	*locale.LocalizedError // for legacy, non-user-facing error usages
```

Comment:
Since not all consumers are using user-facing errors, keep using the localized errors from the buildplanner for those consumers. The user-facing error rationalizer in this PR is making use of `Type`.

Reply:
I believe that for this to be used this type needs an `Error()` function that returns the localized error message. Unless there is another way to surface this that I am missing.

Reply:
I believe it inherits `locale.LocalizedError`'s `Error()` function. Otherwise there would be a compiler error that you cannot use a `CommitError` struct as an `error` type.

Reply:
Ahh yes, you're right!